src/pudl/transform/ferc1.py

```diff
@@ -6160,13 +6218,31 @@ def ferc1_transform_asset(**kwargs: dict[str, pd.DataFrame]) -> pd.DataFrame:
             )
 
         raw_dbf = pd.concat(
-            [df for key, df in kwargs.items() if key.startswith("raw_dbf__")]
+            [df for key, df in kwargs.items() if key.startswith("raw_ferc1_dbf__")]
         )
+        raw_xbrls = {
+            tn: filter_for_freshest_data_xbrl(
+                df=df,
+                primary_keys=_get_primary_key(tn.removeprefix("raw_ferc1_xbrl__")),
+            )
+            for tn, df in kwargs.items()
+            if tn.startswith("raw_ferc1_xbrl__")
+        }
+        for raw_xbrl_table_name in listify(TABLE_NAME_MAP_FERC1[table_name]["xbrl"]):
+            if (
+                raw_xbrls[f"raw_ferc1_xbrl__{raw_xbrl_table_name}_instant"].empty
+                and raw_xbrls[f"raw_ferc1_xbrl__{raw_xbrl_table_name}_duration"].empty
+            ):
+                raise AssertionError(
+                    "We expect there to be no raw xbrl tables that have neither instant or duration "
```

Comment:
nit: `f"{raw_xbrl_table_name} has neither instant nor duration tables. Is it spelled correctly in pudl.extract.ferc1.TABLE_NAME_MAP_FERC1?"` is a bit more information dense